django/utils/crypto.py

```diff
@@ -21,17 +26,20 @@ def salted_hmac(key_salt, value, secret=None):
 
     key_salt = force_bytes(key_salt)
     secret = force_bytes(secret)
+    try:
+        algo = getattr(hashlib, algo or 'sha1')
+    except AttributeError:
+        raise InvalidAlgorithm("'%s' is not an algorithm name accepted by the hashlib module")
```

Comment:
You referred to `hashlib.new()` in the docstring, but then did the `getattr()` stuff here. Perhaps we should assign to a new `hasher` variable too:

```python
hasher = hashlib.new(algorithm)
```

I'm not sure it is necessary to provide our own `InvalidAlgorithm` exception. I think the default is clear enough:

```
>>> hashlib.new('invalid')
Traceback (most recent call last):
  File "/usr/lib/python3.8/hashlib.py", line 157, in __hash_new
    return _hashlib.new(name, data)
ValueError: unsupported hash type

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/hashlib.py", line 163, in __hash_new
    return __get_builtin_constructor(name)(data)
  File "/usr/lib/python3.8/hashlib.py", line 120, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type invalid
```

(Obviously the context will be better if raised from this location.)